core/domain/collection_domain.py

```diff
@@ -435,16 +592,12 @@ def update_collection_contents_from_model(
         versioned_collection_contents['collection_contents'] = conversion_fn(
             versioned_collection_contents['collection_contents'])
 
-    @property
-    def skills(self):
-        """The skills of a collection are made up of all prerequisite and
-        acquired skills of each exploration that is part of this collection.
-        This returns a sorted list of all the skills of the collection.
-        """
-        unique_skills = set()
-        for node in self.nodes:
-            unique_skills.update(node.skills)
-        return sorted(unique_skills)
+    def human_readable_skill_names(self):
+        """Returns a list of skill names of all the skills in the collection."""
+        skill_names = set()
+        for _, skill in self.skills.iteritems():
+            skill_names.update(skill.name)
```

Comment:
Shouldn't this be ``skill_names.add(skill.name)``? Do we have tests for this method?

Also, should we allow multiple skills to be present with the same name? If not, this method can be simplified to:

``return [skill.name for skill in self.skills.values()]``

In which case we should also update the validate() method to ensure no skill names are duplicated.